lib/min_heap.rb

```diff
@@ -1,73 +1,110 @@
 class HeapNode
   attr_reader :key, :value
-
+  
   def initialize(key, value)
     @key = key
     @value = value
   end
 end
 
 class MinHeap
-
+  
   def initialize
     @store = []
   end
-
+  
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: I think O(log N)
+  # Space Complexity: O(1) - I think this is for all space regardless of operation because it's not necessarily
+  #growing all that much in size?
   def add(key, value = key)
```

Comment:
Since `heap_up` is recursive you need to account for the system stack in the space complexity.  So Space complexity is O(log n).  Otherwise well done.